nav2_smac_planner/src/analytic_expansion.cpp

```diff
@@ -249,6 +255,13 @@ void AnalyticExpansion<NodeT>::cleanNode(const NodePtr & /*expanded_nodes*/)
 {
 }
 
+template<typename NodeT>
+void AnalyticExpansion<NodeT>::cleanupDetachedNodes() {
+  for (auto &node : _detached_nodes)
```

Comment:
If these are `unique_ptr`'s then they will auto delete when you `clear()`. In Nav2, we're trying (as much as reasonably possible) to use smart pointers instead of raw memory in case of later changes to the system to avoid leaks. 

If you could change this to `unique_ptr` or another smart pointer type, I'd be happy to merge this tomorrow.